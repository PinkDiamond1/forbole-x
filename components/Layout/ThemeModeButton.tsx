import { IconButton } from '@material-ui/core'
import React from 'react'
import useStyles from './styles'
import DarkModeIcon from '../../assets/images/icons/icon_dark_mode.svg'
import LightModeIcon from '../../assets/images/icons/icon_light_mode.svg'
import { useSettingsContext } from '../../contexts/SettingsContext'
import useIconProps from '../../misc/useIconProps'

interface ThemeModeButtonProps {}

const ThemeModeButton: React.FC<ThemeModeButtonProps> = () => {
  const { theme, setTheme } = useSettingsContext()
  const iconProps = useIconProps(3)
  const classes = useStyles()
  const isDarkMode = theme === 'dark'
  return (
    <IconButton
      onClick={() => setTheme(isDarkMode ? 'light' : 'dark')}
      className={classes.navBarButton}
    >
      {isDarkMode ? <LightModeIcon {...iconProps} /> : <DarkModeIcon {...iconProps} />}
    </IconButton>
  )
}

export default React.memo(ThemeModeButton)
