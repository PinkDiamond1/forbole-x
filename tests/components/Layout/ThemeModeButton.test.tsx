import React from 'react'
import renderer from 'react-test-renderer'
import ThemeModeButton from '../../../components/Layout/ThemeModeButton'

const mockThemeContext = {
  theme: 'light',
  setTheme: jest.fn(),
}

jest.mock('../../../contexts/SettingsContext', () => ({
  useSettingsContext: () => mockThemeContext,
}))

describe('component: ThemeModeButton', () => {
  it('renders light theme correctly', () => {
    mockThemeContext.theme = 'light'
    const component = renderer.create(<ThemeModeButton width={10} height={10} fill="#000" />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders dark theme correctly', () => {
    mockThemeContext.theme = 'dark'
    const component = renderer.create(<ThemeModeButton width={10} height={10} fill="#000" />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('sets theme to dark on click', () => {
    mockThemeContext.theme = 'light'
    const component = renderer.create(<ThemeModeButton width={10} height={10} fill="#000" />)
    component.root.findByType('button').props.onClick()
    expect(mockThemeContext.setTheme).toBeCalledWith('dark')
  })
  it('sets theme to light on click', () => {
    mockThemeContext.theme = 'dark'
    const component = renderer.create(<ThemeModeButton width={10} height={10} fill="#000" />)
    component.root.findByType('button').props.onClick()
    expect(mockThemeContext.setTheme).toBeCalledWith('light')
  })
})

afterEach(() => {
  jest.clearAllMocks()
})
