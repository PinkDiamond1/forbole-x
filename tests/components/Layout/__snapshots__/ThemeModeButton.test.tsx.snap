// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: Layout - ThemeModeButton renders dark theme correctly 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root HookGlobalStyles-navBarButton-9"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiIconButton-label"
  >
    <svg
      data-file-name="SvgIcon_light_mode"
      fill="#e0e0e0"
      height={24}
      width={24}
    />
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;

exports[`component: Layout - ThemeModeButton renders light theme correctly 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root HookGlobalStyles-navBarButton-9"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiIconButton-label"
  >
    <svg
      data-file-name="SvgIcon_dark_mode"
      fill="#e0e0e0"
      height={24}
      width={24}
    />
  </span>
</button>
`;
